sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -169,7 +170,7 @@ public PBegin begin() {
    * @see #apply(String, PTransform)
    */
   public <OutputT extends POutput> OutputT apply(
-      PTransform<? super PBegin, OutputT> root) {
+      PTransform<? super PBegin, OutputT> root) throws ValidationException {
```

Comment:
Well, the only reason to have a `ValidationException` is that in generally a caller should treat it as alternative return code path, so "all" validation errors should proceed down that path. I agree this seems cumbersome. Your basic `main()` might just log, or a more extensive framework might put validation errors in a particular bucket separate from miscellaneous errors. But the line is blurry between validation errors and other occurrences.

@jkff you make a very good point that we probably don't need this exception. When a checked exception like `IOException` comes back from some process during `validate()` probably it should just be re-thrown as an `IllegalArgumentException` or `IllegalStateException` with a new error message explaining the context for the failure.